llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -765,6 +765,16 @@ def unmerge_merge : GICombineRule<
   (apply [{ Helper.applyCombineUnmergeMergeToPlainValues(*${d}, ${info}); }])
 >;
 
+// Fold (unmerge(merge x, y, z)) -> z, y, z.
+def shufflevector_merge_matchinfo : GIDefMatchData<"SmallVector<Register, 8>">;
+def shufflevector_merge : GICombineRule<
+  (defs root:$d, shufflevector_merge_matchinfo:$info),
+  (match (wip_match_opcode G_SHUFFLE_VECTOR): $d,
```

Comment:
A recent move in LLVM is that you shouldn't use `wip_match_opcode` anymore since it slows down compilation. 

https://llvm.org/docs/GlobalISel/MIRPatterns.html#gallery